### PR TITLE
Upgrade to graphql-java lib 17.3

### PR DESCRIPTION
**Pull request by CNAChino**
# What Changed
- upgrade graphql-java to 17.3
- refactor, update affecte classes
- add integration tests to ensure that proxying logic is working for new capabilities

# Why

Todo:

- [x] Add tests
- [ ] Add docs

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -37,7 +37,7 @@
     <awssdk2.version>2.7.2</awssdk2.version>
     <graphql-sdl-version>3.0.0</graphql-sdl-version>
     <xtextVersion>2.25.0</xtextVersion>
-    <graphQLVersion>14.1</graphQLVersion>
+    <graphQLVersion>15.0</graphQLVersion>
```
Comment (CNAChino) [inline pom.xml:40]: this will be 17.3 once the pr is completed

File: src/main/java/com/intuit/graphql/orchestrator/utils/ValueUtil.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import graphql.schema.GraphQLType;
+import graphql.schema.PropertyDataFetcherHelper;
+import graphql.util.FpKit;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The code from this is from graphql-java library AstValueHelper
```
Comment (ashpak-shaikh) [inline src/main/java/com/intuit/graphql/orchestrator/utils/ValueUtil.java:33]: Have they removed this ?

**Comment by ashpak-shaikh** [comment]
have we taken the latest graphql-xtext as well?

File: src/main/java/com/intuit/graphql/orchestrator/utils/ValueUtil.java (hunk 2 above)
Comment (CNAChino) [reply src/main/java/com/intuit/graphql/orchestrator/utils/ValueUtil.java:33]: yup.  I was not able to find a replacement.  ValueResolver.java looks like does not have something similar but will revisit that class

**Comment by CNAChino** [comment]
> have we taken the latest graphql-xtext as well?

not yet but will do

File: src/main/java/com/intuit/graphql/orchestrator/utils/XtextToGraphQLJavaVisitor.java [hunk 3]
```diff
@@ -75,8 +75,11 @@
   public final Map<String, GraphQLDirective> directiveDefinitions;
 
   static {
-    STANDARD_SCALAR_TYPES = ScalarInfo.STANDARD_SCALARS.stream()
+    STANDARD_SCALAR_TYPES = ScalarInfo.GRAPHQL_SPECIFICATION_SCALARS.stream()
         .collect(Collectors.toMap(GraphQLScalarType::getName, Function.identity()));
+
+    STANDARD_SCALAR_TYPES.putAll(ExtendedScalarsSupport.GRAPHQL_EXTENDED_SCALARS.stream()
```
Comment (CNAChino) [inline src/main/java/com/intuit/graphql/orchestrator/utils/XtextToGraphQLJavaVisitor.java:81]: looks like we don't need this.  scalars should be defined in sdl

File: src/main/java/com/intuit/graphql/orchestrator/utils/ValueUtil.java (hunk 2 above)
Comment (kmoore-intuit) [reply src/main/java/com/intuit/graphql/orchestrator/utils/ValueUtil.java:33]: you may be able to use the Parser.parseValue that is what they changed to

Comment (kmoore-intuit) [reply src/main/java/com/intuit/graphql/orchestrator/utils/ValueUtil.java:33]: I was intrigued so I looked a little into it, the method you may be looking for is ValuesResolver.valueToLiteral() but value would have to be converted to a InputValueWithState

Comment (ashpak-shaikh) [reply src/main/java/com/intuit/graphql/orchestrator/utils/ValueUtil.java:33]: just to close the loop on this. we found out the replacement methods for this class and @CNAChino 
should be adding those changes in the next commit

Comment (ashpak-shaikh) [reply src/main/java/com/intuit/graphql/orchestrator/utils/ValueUtil.java:33]: Should we remove this class

File: src/test/groovy/com/intuit/graphql/orchestrator/integration/InterfaceImplementInterfaceSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,124 @@
+package com.intuit.graphql.orchestrator.integration
+
+
+import graphql.ExecutionInput
+import graphql.ExecutionResult
+import helpers.BaseIntegrationTestSpecification
+import spock.lang.Subject
+
+class InterfaceImplementInterfaceSpec extends BaseIntegrationTestSpecification {
```
Comment (CNAChino) [inline src/test/groovy/com/intuit/graphql/orchestrator/integration/InterfaceImplementInterfaceSpec.groovy:9]: look into XtextGraphQLVisitor.caseInterfaceTypeDefinition

File: src/test/groovy/com/intuit/graphql/orchestrator/integration/DirectiveSpec.groovy [hunk 5]
```diff
@@ -0,0 +1,33 @@
+package com.intuit.graphql.orchestrator.integration
+
+import com.intuit.graphql.orchestrator.schema.SchemaParseException
+import com.intuit.graphql.orchestrator.stitching.SchemaStitcher
+import helpers.BaseIntegrationTestSpecification
+
+class DirectiveSpec extends BaseIntegrationTestSpecification {
```
Comment (CNAChino) [inline src/test/groovy/com/intuit/graphql/orchestrator/integration/DirectiveSpec.groovy:7]: add repeatable directive tests

File: src/main/java/com/intuit/graphql/orchestrator/utils/ValueUtil.java (hunk 2 above)
Comment (CNAChino) [reply src/main/java/com/intuit/graphql/orchestrator/utils/ValueUtil.java:33]: The solution is to use `ValuesResolver.valueToLiteral(inputValueWithState, type)`.   And use `InputValueWithState.newExternalValue()`.  `InputValueWithState.newInternalValue()` will be removed soon.

Comment (CNAChino) [reply src/main/java/com/intuit/graphql/orchestrator/utils/ValueUtil.java:33]: > I was intrigued so I looked a little into it, the method you may be looking for is ValuesResolver.valueToLiteral() but value would have to be converted to a InputValueWithState

👍

Comment (CNAChino) [reply src/main/java/com/intuit/graphql/orchestrator/utils/ValueUtil.java:33]: > Should we remove this class

yes

**Review by bhavinshah7 — APPROVED** [review]

**Pull request merged**